cmd/fs-v1.go

```diff
@@ -72,15 +72,83 @@ func initMetaVolumeFS(fsPath, fsUUID string) error {
 
 }
 
+// Attempts to migrate old object metadata files to newer format
+//
+// i.e
+//    -------------------------------------------------------
+//    .minio.sys/buckets/<bucket_name>/<object_path>/fs.json
+//    -------------------------------------------------------
+//    .minio.sys/buckets/<bucket_name>/objects/<object_path>/fs.json
+//    -------------------------------------------------------
+//
+func migrateFSObjectMetadataDir(fsPath, fsUUID string) error {
```

Comment:
Copying each fs.json can take ages. We don't need to do that. Renames are atomic in nature so either it's do or die.. if we print a message about migration then all of this works out nicely.